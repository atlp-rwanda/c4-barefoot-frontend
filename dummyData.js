export const user = [
  {
    "first_name":"M",
    "last_name":"JAckson",
    "email":"jackson.novelty@gmail.com",
    "occupation":"Programming",
    "username":"jacks",
    "bio":"Software Developer",
    "address":"kk 509 st",
    "language":"Kinyarwanda",
    "password":"12345678",
    "profile_picture":"https://res.cloudinary.com/mjackson/image/upload/v1609304524/Screen_Shot_2020-12-29_at_08.21.43_qe7pem.png"
  }
]

export const signupState = {
    requesting: false,
    errorOpen: false,
    erro: ""
  }
export const locationsPayload = [
    {
      id: "c6028e0d-ef88-4693-ab49-f37669891724",
      LocationName: "Kigali",
      country: "Rwanda",
      description: "The land of a thousand hills",
      link: "https://www.nationalgeographic.com/content/dam/travel/Guide-Pages/africa/downtown-kigali-rwanda.adapt.1900.1.jpg",
    }
  ]

export const accommodationsPayload = [
    {
      
        id: "0ce36391-2c08-3074-bddb-a4ea8cccbbc5",
        country: "Rwanda",
        city: "Kigali",
        title: "Marriot Hotel",
        description: "A serene environment for relaxation",
        photos: "https://cache.marriott.com/marriottassets/marriott/KGLMC/kglmc-exterior-0030-hor-feat.jpg",
    
    }
  ]

  export const accommodationState = {
    pending: false,
    accommodations: accommodationsPayload,
    error: null
  }

  export const locationState = {
    pending: false,
    locations: locationsPayload,
    error: null
  }

