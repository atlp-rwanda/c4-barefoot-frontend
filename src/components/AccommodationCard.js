import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Card, CardActionArea, CardActions, CardContent, CardMedia, Button, Typography} from '@material-ui/core';
import { Place } from '@material-ui/icons'
import { Skeleton } from '@material-ui/lab'
import { connect } from 'react-redux'



const useStyles = makeStyles((theme) => ({
  root: {
    maxWidth: 345,
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between'
  },
  media: {
    height: 140
  },
}));

function Accommodations(props) {
  
  const classes = useStyles();

  return (
    <Card className={classes.root}>
      <CardActionArea>
        { props.pending ? 
        (<Skeleton variant='rect' animation="wave" className={classes.media} />)
        :
        (<CardMedia
          className={classes.media}
          image={props.accommodation.photos}
          title={props.accommodation.title}
        />)}
        <CardContent>
          <Typography gutterBottom variant="h5" component="h2">
           { props.pending ? (<Skeleton animation="wave" width="50%"/>)
           : (props.accommodation.title)}
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p">
            { props.pending ? 
            (<React.Fragment>
              <Skeleton animation="wave" />
              <Skeleton animation="wave" width='80%' />
            </React.Fragment>)
            : (props.accommodation.description)}
          </Typography>
        </CardContent>
      </CardActionArea>
      <CardActions>
      {props.pending ? (<Skeleton animation='wave' width='20%' height='20px' />) :
        <Button size="small" color="primary" startIcon={<Place />}> {props.accommodation.city} </Button>
      }
      </CardActions>
    </Card>
  );
}

const mapStateToProps = state => ({
  pending: state.fetchAccommodations.pending,
})

export default connect(mapStateToProps, null)(Accommodations);
