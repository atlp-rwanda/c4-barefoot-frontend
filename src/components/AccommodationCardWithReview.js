import React, {useEffect, useState} from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Card, CardActionArea, CardActions, CardContent, CardMedia, Button, Typography, Checkbox} from '@material-ui/core';
import { Place } from '@material-ui/icons'
import { Skeleton } from '@material-ui/lab'
import colors from './colors';
import Ratings from './RatingStars';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom'
import {getBookings} from "../redux/actions/bookAccommodationAction";

const useStyles = makeStyles((theme) => ({
  root: {
    maxWidth: 345,
    height: 360,
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between'
  },
  media: {
    height: 140
  },
  checkbox:{
      display: 'block',
      position:'absolute',
      color: 'secondary',
      right: 0,
      fontSize:'25px'
  },
  cardActions:{
      display: 'flex',
      flexDirection:'column',
      alignItems:'flex-start',
  },
  reviews:{
      fontSize: '14px',
      color: colors.primary100,
  },
  cardContent:{
    overflow: 'hidden'
  },
  titleText:{
    [theme.breakpoints.down('sm')]:{
      fontSize:'18px',
    }
  },
  rateAndReviewLinks: {
    display: 'flex',
    justifyContent: 'space-between',
    padding:'5px'
  }
}));

function Accommodations(props) {
    const classes = useStyles();
    const open = false; 
    let [check,setCheck]=useState(null);
    const handleSelection = (event) =>{
      setCheck(event.target.id);
      console.log(event.target.name);
      props.selectAccommodation(event.target.id);
     // props.getAccommodation(event.target.name);
     console.log(props)
      props.getTemperature(event.target.name);
      props.getAccommodationAminity(event.target.id);

  };
  useEffect(() => {
    props.getBookings()
    // console.log(props.reviews)
  },[])
    const handleViewMore =() =>{
      return props.openModalAction({open: true, data: props.accommodation});
    }


  return (
    <Card className={classes.root} >
      {(props.pending ? 
        <CardActionArea>
          <Skeleton animation="wave" variant="rect" className={classes.media} />
          <CardContent className={classes.cardContent} >
            <Skeleton animation="wave" height={30} width="60%" />
            <Skeleton animation="wave" height={10} width="80%" />
          </CardContent>
          <CardActions className={classes.cardActions}>
            <Skeleton animation="wave" height={30} width="60%" />
          </CardActions>
        </CardActionArea>
        :
      <>
      <CardActionArea>
        <Checkbox
        onChange={handleSelection}
        checked={props.accommodationn.id===props.selectedAccommodation?true:false}
        id={props.accommodationn.id}
        name={props.accommodationn.city}
        className={classes.checkbox}/>
        <CardMedia
          onClick={handleViewMore}
          className={classes.media}
          image={props.accommodationn.photos}
              // image={image}
          title={props.accommodationn.title}
        />
        <CardContent onClick={handleViewMore} className={classes.cardContent} >
          <Typography gutterBottom variant="h5" component="h2" className={classes.titleText}>
           {props.accommodationn.title}
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p" noWrap >
            {props.accommodationn.description}
          </Typography>
        </CardContent>
          </CardActionArea>
          {props.accommodation.bookedAccommodation != undefined ? (
            <CardActions className={classes.cardActions}>
               <Button size="small" color="primary" onClick={handleViewMore} startIcon={<Place color="secondary" />}> {props.accommodationn.state}, {props.accommodationn.city} </Button>
              <div className={classes.rateAndReviewLinks}>
                <div>
                <Link  to={'/review/'+props.accommodationn.id}>
                   <Ratings highratings={3} id={props.accommodationn.id} />
                 </Link>
                </div>
                <div>
                <Typography className={classes.reviews}>
             <Link  to={'/reviews/'+props.accommodationn.id} className={classes.links}  style={{ textDecoration: 'none' }}>
                   Reviews
                   </Link>
             </Typography>
                </div>
              </div>
            
                 
            
           </CardActions>
          ):( <CardActions className={classes.cardActions}>
            <Button size="small" color="primary" onClick={handleViewMore} startIcon={<Place color="secondary" />}> {props.accommodationn.state}, {props.accommodationn.city} </Button>
                
                  <Ratings highratings={3} id={props.accommodationn.id} />
                
            <Typography className={classes.reviews}>
                  Reviews
            </Typography>
          </CardActions>)}
     
      </>
      )}
    </Card>
  );
}

const mapStateToProps = state => ({
  accommodation:state.bookedAccommodations
})

export default connect(mapStateToProps, {getBookings}) (Accommodations);

