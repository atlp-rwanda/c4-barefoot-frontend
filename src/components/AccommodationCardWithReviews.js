import React, {useEffect, useState} from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Card, CardActionArea, CardActions, CardContent, CardMedia, Button, Typography, Checkbox} from '@material-ui/core';
import { Place } from '@material-ui/icons'
import { Skeleton } from '@material-ui/lab'
import colors from './colors';
import Ratings from './RatingStars';

const useStyles = makeStyles((theme) => ({
  root: {
    maxWidth: 345,
    height: 360,
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between'
  },
  media: {
    height: 140
  },
  checkbox:{
      display: 'block',
      position:'absolute',
      color: 'secondary',
      right: 0,
      fontSize:'25px'
  },
  cardActions:{
      display: 'flex',
      flexDirection:'column',
      alignItems:'flex-start',
  },
  reviews:{
      fontSize: '14px',
      color: colors.primary100,
  },
  cardContent:{
    overflow: 'hidden'
  },
  titleText:{
    [theme.breakpoints.down('sm')]:{
      fontSize:'18px',
    }
  }
}));

function Accommodations(props) {
    const classes = useStyles();
    const open = false; 
    const handleSelection = (event) =>{
      const accommodation={
        selected:props.accommodation,
        checked: event.target.checked,
      };
    
      return props.selectAccommodationAction(accommodation);

    }    
    
    
    let check=false;
    props.travelRequest.selectedAccommodation.map( selected =>{
      selected.id === props.accommodation.id ? check=true : null
    })
    const handleViewMore =() =>{
      return props.openModalAction({open: true, data: props.accommodation});
    }


  return (
    <Card className={classes.root} >
      {(props.pending ? 
        <CardActionArea>
          <Skeleton animation="wave" variant="rect" className={classes.media} />
          <CardContent className={classes.cardContent} >
            <Skeleton animation="wave" height={30} width="60%" />
            <Skeleton animation="wave" height={10} width="80%" />
          </CardContent>
          <CardActions className={classes.cardActions}>
            <Skeleton animation="wave" height={30} width="60%" />
          </CardActions>
        </CardActionArea>
        :
      <>
      <CardActionArea>
        <Checkbox
        onChange={handleSelection}
        checked={check}
        className={classes.checkbox}/>
        <CardMedia
          onClick={handleViewMore}
          className={classes.media}
          image={props.accommodation.photos}
          title={props.accommodation.title}
        />
        <CardContent onClick={handleViewMore} className={classes.cardContent} >
          <Typography gutterBottom variant="h5" component="h2" className={classes.titleText}>
           {props.accommodation.title}
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p" noWrap >
            {props.accommodation.description}
          </Typography>
        </CardContent>
      </CardActionArea>
      <CardActions className={classes.cardActions}>
        <Button size="small" color="primary" onClick={handleViewMore} startIcon={<Place color="secondary" />}> {props.accommodation.state}, {props.accommodation.city} </Button>
        <Ratings highRating={3} />
        <Typography className={classes.reviews}>
            25 Reviews
        </Typography>
      </CardActions>
      </>
      )}
    </Card>
  );
}

export default Accommodations;
