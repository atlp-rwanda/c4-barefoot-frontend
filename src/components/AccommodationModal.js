import React from 'react';
import {Modal, Fade, Typography, CardMedia,Grid} from '@material-ui/core';
import {makeStyles} from '@material-ui/core/styles';

const useStyles = makeStyles((theme) =>({
    modal:{
        display: 'Flex',
        alignItems: 'center',
        justifyContent: 'center'
    },
    loader:{
        backgroundColor: theme.palette.background.paper,
        boxShadow: theme.shadows[5],
        outline: 'none',
        padding: theme.spacing(2, 4, 3),
        textAlign: 'left',
        width:'50%',
        height: 'auto',
        [theme.breakpoints.down('sm')]:{
            width:'70%'
        }
    },
    media:{
        height:"300px",
        [theme.breakpoints.down('sm')]:{
            height:"200px"
        }
    },
    otherInfo:{
        borderTop: '1px solid darkCyan',
        justifyContent: 'flex-start',
        [theme.breakpoints.down('sm')]:{
            fontSize:"18px"
        }
    }

}));

function AccommodationModal (props) {

    const classes = useStyles();
    const handleCloseModal = () =>{
        return props.openModalAction({open:false,data:props.data});
    }
    return (
        <React.Fragment>
            <Modal
            className={classes.modal}
            open={props.open}
            closeAfterTransition
            onClick={handleCloseModal}
            >
                <Fade in={props.open}>
                    <div className={classes.loader}>
                        <CardMedia
                        className={classes.media} 
                        image={props.data.photos}
                        title={props.data.title}
                        />
                        <Typography variant="h5" component="h5">
                            {props.data.title}
                        </Typography>
                        <Typography
                        variant="subtitle1"
                        >{props.data.description}</Typography>
                        <Grid item className={classes.otherInfo}>
                            <Typography variant="subtitle2">Country: {props.data.country}</Typography>
                            <Typography variant="subtitle2">City: {props.data.city}</Typography>
                            <Typography variant="subtitle2">State: {props.data.state}</Typography>
                            <Typography variant="subtitle2" style={{marginBottom:'15px'}}>Street: {props.data.streetAddress}</Typography>

                            <Typography variant="subtitle2">Type of Property: {props.data.propertyType}</Typography>
                            <Typography variant="subtitle2">Number of Rooms: {props.data.numberOfRooms}</Typography>
                            <Typography variant="subtitle2">Type of Beds: {props.data.typeOfBed}</Typography>
                        </Grid>
                    </div>
                </Fade>
            </Modal>
        </React.Fragment>
    )
}

export default AccommodationModal;