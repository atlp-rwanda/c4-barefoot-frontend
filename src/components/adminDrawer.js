import React, {useState} from 'react'
import { makeStyles, Drawer, List, Divider, ListItem, ListItemIcon, ListItemText, IconButton, Box, Avatar, Typography} from '@material-ui/core';
import { PlusOne, Home, People, Delete, Settings, Menu, AccountCircle } from '@material-ui/icons'
import { Link } from 'react-router-dom'


const useStyles = makeStyles(theme => ({
  list: {
    width: 250,
  },
  menuIcon: {
    color: 'white'
  },
  linkText: {
    textDecoration: `none`,
    textTransform: `uppercase`,
    color: 'black'
  },
  accountInfo: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    justifyContent: 'space-evenly',
    height: 80
  },
  paper: {
    backgroundColor: '#EAF4FB'
  },
  listIcons: {
    color: theme.palette.primary.main,
  }
}));

export default function TemporaryDrawer() {
  const classes = useStyles();
  const [state, setState] = useState({left: false});

  const firstLinks = [
    {title: 'Home', path: '/adminHome', icon: <Home className={classes.listIcons}/>},
    {title: 'Create Roles', path: '/roles', icon: <PlusOne className={classes.listIcons}/>},
    {title: 'Set permissions', path:'/permissions', icon: <Settings className={classes.listIcons} />}
  ]

  const toggleDrawer = (anchor, open) => (event) => {
    if (event.type === 'keydown' && (event.key === 'Tab' || event.key === 'Shift')) {
      return;
    }

    setState({ [anchor]: open });
  };

  const list = (anchor) => (
    <div
      className={classes.list}
      role="presentation"
      onClick={toggleDrawer(anchor, false)}
      onKeyDown={toggleDrawer(anchor, false)}
    >
      <Box className={classes.accountInfo}>
        <Avatar className={classes.paper}>
          <AccountCircle fontSize='large' className={classes.listIcons}/>
        </Avatar>
        <Typography>Admin Name</Typography>
      </Box>
      <Divider/>
      <List>
        {firstLinks.map(({title, path, icon}) => (
          <Link to={path} key={title} className={classes.linkText} >
            <ListItem button>
              <ListItemIcon> {icon} </ListItemIcon>
              <ListItemText primary={title} />
            </ListItem>
          </Link>
        ))}
      </List>
      <Divider/>
      <List>
        <Link to='/users' key='List users' className={classes.linkText}>
          <ListItem button>
            <ListItemIcon> <People className={classes.listIcons}/> </ListItemIcon>
            <ListItemText primary= 'List users' />
          </ListItem>
        </Link>
      </List>
      <Divider/>
      <List>
        <Link to='/trash' key='Trash' className={classes.linkText}>
          <ListItem button>
            <ListItemIcon> <Delete className={classes.listIcons}/> </ListItemIcon>
            <ListItemText primary= 'Trash' />
          </ListItem>
        </Link>
      </List>
    </div>
  )

  return(
    <div>
      
        <React.Fragment>
          <IconButton edge='start' onClick={toggleDrawer('left', true)}>
            <Menu fontSize='large' className = {classes.menuIcon} />
          </IconButton>
          <Drawer 
            classes = {{paper: classes.paper}}
            anchor='left' 
            open={state.left} 
            onOpen={toggleDrawer('left', true)}
            onClose={toggleDrawer('left', false)}>
            {list('left')}
          </Drawer>
        </React.Fragment>
      
    </div>
  )
}