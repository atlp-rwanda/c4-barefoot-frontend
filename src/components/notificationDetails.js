import React from 'react';
import { useParams} from "react-router-dom";
import {connect} from "react-redux";
import {readTravelRequestInfo} from "../redux/actions/notificationAction";


const NotificationDetails=(props)=>{
    React.useEffect(()=>{
        props.readTravelRequestInfo()
    })
    const request=props.request
    console.log(request)
    return(
        <div>
           {
               request&&
               <div>
                   <Typography>{request.userId}</Typography>
               </div>
           }
        </div>
    )
}
const mapStateToProps=(state)=>({
   request:state.notifications.travelRequest
})
export default connect(mapStateToProps,{readTravelRequestInfo})(NotificationDetails);