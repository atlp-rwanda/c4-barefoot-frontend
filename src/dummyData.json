{
    "locations": [
      {
        "city": "capetown",
        "description": "This is one of best cities in Africa and you can also find good accomodations.",
        "image": "https://freewalkingtourscapetown.co.za/wp-content/uploads/2018/05/cropped-cape-town-header-pic-1.png"
      },
      {
        "city": "capetown",
        "description": "This is one of best cities in Africa and you can also find good accomodations.",
        "image": "https://freewalkingtourscapetown.co.za/wp-content/uploads/2018/05/cropped-cape-town-header-pic-1.png"
      },
      {
        "city": "capetown",
        "description": "This is one of best cities in Africa and you can also find good accomodations.",
        "image": "https://freewalkingtourscapetown.co.za/wp-content/uploads/2018/05/cropped-cape-town-header-pic-1.png"
      }
    ],
    "accommodations": [
      {
        "title": "Marriott Hotel",
        "location": "Kigali",
        "description": "This is one of best cities in Africa and you can also find good accomodations.",
        "image": "https://cache.marriott.com/marriottassets/marriott/KGLMC/kglmc-exterior-0030-hor-feat.jpg"
      },
      {
        "title": "Serena Hotel",
        "location": "Kigali",
        "description": "This is one of best cities in Africa and you can also find good accomodations.",
        "image": "https://cache.marriott.com/marriottassets/marriott/KGLMC/kglmc-exterior-0030-hor-feat.jpg"
      },
      {
        "title": "Marriott Hotel",
        "location": "Kigali",
        "description": "This is one of best cities in Africa and you can also find good accomodations.",
        "image": "https://cache.marriott.com/marriottassets/marriott/KGLMC/kglmc-exterior-0030-hor-feat.jpg"
      }
    ]
}
