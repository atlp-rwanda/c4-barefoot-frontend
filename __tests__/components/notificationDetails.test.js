import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { NotificationDetails } from '../../src/components/notificationDetails';
import { singleTravelPayload } from '../../dummyData';
import { Typography } from '@material-ui/core';

describe('NOTIFICATION DETAILS COMPONENT', ()=>{
    let wrapper;
    const props = {
        match: {
            params: {
                id: '9b1e1ca6-2fc2-42bb-86c0-86a329703393'
            }
        }, 
        getNotifications: jest.fn(),
        readTravelRequestInfo: jest.fn(),
        request: singleTravelPayload
        
    }
    beforeEach(()=>{
        wrapper = shallow(<NotificationDetails {...props}/>);
    })
    it('Should capture the snapshot', ()=>{
        
        const component = toJson(wrapper);
        expect(component).toMatchSnapshot();
    })
    it('Should render the component', ()=>{
        expect(wrapper.find(Typography).length).toEqual(1);
    })
})