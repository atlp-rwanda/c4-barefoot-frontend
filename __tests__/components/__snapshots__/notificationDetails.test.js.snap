// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NOTIFICATION DETAILS COMPONENT Should capture the snapshot 1`] = `
<NotificationDetails
  getNotifications={[MockFunction]}
  match={
    Object {
      "params": Object {
        "id": "9b1e1ca6-2fc2-42bb-86c0-86a329703393",
      },
    }
  }
  readTravelRequestInfo={
    [MockFunction] {
      "calls": Array [
        Array [
          "9b1e1ca6-2fc2-42bb-86c0-86a329703393",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  request={
    Array [
      Object {
        "accomodationInfo": Array [
          Object {
            "city": "Kigali",
            "country": "Rwanda",
            "description": "A serene environment for relaxation",
            "id": "520f2b37-7bac-4490-aa7a-96f15915bcd7",
            "locationID": "c6028e0d-ef88-4693-ab49-f37669891724",
            "numberOfRooms": 0,
            "photos": "image.png",
            "propertyType": "Hostel",
            "state": "Nyarugenge",
            "streetAddress": "KN 22 ST",
            "title": "Kigali Hostels",
            "typeOfBed": "Double Decker",
          },
        ],
        "travelRequestInfo": Object {
          "Trip": Array [
            Object {
              "accommodationId": "520f2b37-7bac-4490-aa7a-96f15915bcd7",
              "createdAt": "2021-03-28T07:02:57.590Z",
              "destination": "biryogo",
              "originCity": "kuryanyuma",
              "reason": "travelling",
              "returnDate": null,
              "travelId": "1a27b15c-4138-43ac-bb64-c0b4e2c919a0",
              "tripDate": "2020-12-11T06:48:48.376Z",
              "tripId": "d81145c6-5778-43b0-aac9-5cf8360f3722",
              "updatedAt": "2021-03-28T07:02:57.590Z",
            },
          ],
          "createdAt": "2021-03-28T07:02:57.389Z",
          "managerId": "fb94de4d-47ff-4079-89e8-b0186c0a3be8",
          "status": "pending",
          "travelId": "1a27b15c-4138-43ac-bb64-c0b4e2c919a0",
          "updatedAt": "2021-03-28T07:02:57.436Z",
          "userId": "0ce36391-2c08-4703-bddb-a4ea8cccbbc5",
        },
        "userInfo": Object {
          "firstName": "Manager",
          "lastName": "One",
          "profilePpicture": "https://www.cobdoglaps.sa.edu.au/wp-content/uploads/2017/11/placeholder-profile-sq.jpg",
          "username": "managerOne",
        },
      },
    ]
  }
>
  <WithStyles(ForwardRef(Typography))>
    <ForwardRef(Typography)
      classes={
        Object {
          "alignCenter": "MuiTypography-alignCenter",
          "alignJustify": "MuiTypography-alignJustify",
          "alignLeft": "MuiTypography-alignLeft",
          "alignRight": "MuiTypography-alignRight",
          "body1": "MuiTypography-body1",
          "body2": "MuiTypography-body2",
          "button": "MuiTypography-button",
          "caption": "MuiTypography-caption",
          "colorError": "MuiTypography-colorError",
          "colorInherit": "MuiTypography-colorInherit",
          "colorPrimary": "MuiTypography-colorPrimary",
          "colorSecondary": "MuiTypography-colorSecondary",
          "colorTextPrimary": "MuiTypography-colorTextPrimary",
          "colorTextSecondary": "MuiTypography-colorTextSecondary",
          "displayBlock": "MuiTypography-displayBlock",
          "displayInline": "MuiTypography-displayInline",
          "gutterBottom": "MuiTypography-gutterBottom",
          "h1": "MuiTypography-h1",
          "h2": "MuiTypography-h2",
          "h3": "MuiTypography-h3",
          "h4": "MuiTypography-h4",
          "h5": "MuiTypography-h5",
          "h6": "MuiTypography-h6",
          "noWrap": "MuiTypography-noWrap",
          "overline": "MuiTypography-overline",
          "paragraph": "MuiTypography-paragraph",
          "root": "MuiTypography-root",
          "srOnly": "MuiTypography-srOnly",
          "subtitle1": "MuiTypography-subtitle1",
          "subtitle2": "MuiTypography-subtitle2",
        }
      }
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Loading
      </p>
    </ForwardRef(Typography)>
  </WithStyles(ForwardRef(Typography))>
</NotificationDetails>
`;
